MovingBlocks/Terasology
# feat: parallelize tests in `engine`

### Contains

1. `Junit 5` run tests in parallel
2. `Javassist` update to 3.27.0-GA - supports java 11.

All tests not marked via `TestResourceLock#CORE_REGISTRY` runs in parallel. (simple tests)



### How to test

1. run `gradlew :engine-tests:test`
    tests should passed.
2. run `engine-tests`'s tests from IDEA

### Outstanding before merging

- [X] parallelize MTE tests. (CANCEL. PR not affects module testing)
- [X] fix GameNameTest. - flacky
